DiamondLightSource/SynchWeb
# LIMS-1238: Speed up query to list persons on visit

**JIRA ticket**: [LIMS-1238](https://jira.diamond.ac.uk/browse/LIMS-1238)

**Summary**:

Speed up the query to show the users on a visit. Also tidy up some of the code around it.

**Changes**:
- The `$sid` option was looking at an invalid table name, so presumably not used and now deleted
- Remove the `addPersonOrProposalSearch` function and move it's code into `getUsers`, add the joins to `ProposalHasPerson` and `LabContact` at the same time rather than always joining them.
- Fix some typos in comments
- Remove unnecessary `concat`'s and `lower`'s in the search function
- If a visit is specified, use regex to split into proposal code, proposal number and visit number and then match those separately, rather than matching on the `concat` of all 3.

**To test**:
- Go to the Synchweb home page, hover over the little persons icon next to each visit, check the people displayed match that shown on prod, but faster
- Go into a visit, hover over the "Users" button at the top, check the same information is displayed
- Modify line 266 of client/src/js/modules/admin/routes.js from `manage_users` to a permission you have (eg `manage_vusers`)
- Go to /admin/users, you should see your name and fedid, but you can search to see more people
- Check the search bar works on given name, family name and login
- Click on a person, check you can see their basic info
